signcrypt_open.go

```diff
@@ -264,9 +269,13 @@ func (sos *signcryptOpenStream) processSigncryptionBlock(bl *signcryptionBlock)
 	signatureInput = append(signatureInput, nonce[:]...)
 	signatureInput = append(signatureInput, plaintextHash[:]...)
 
-	sigErr := sos.signingPublicKey.Verify(signatureInput, detachedSig[:])
-	if sigErr != nil {
-		return nil, ErrBadSignature
+	// Handle anonymous sender mode by skipping signature verification. By
+	// convention the signature bytes are all zeroes, but here we ignore them.
+	if sos.signingPublicKey != nil {
```

Comment:
Actually, I want to change this. It would be terrible if an accidentally-nil public key got interpreted as "ignore all the signatures". (Even if in theory the app should fail to report a sender.)